python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1303,8 +1319,12 @@ def _ensure_persisted_execution_plan_snapshot(
         return execution_plan_snapshot_id
 
     def _log_asset_planned_events(
-        self, dagster_run: DagsterRun, execution_plan_snapshot: "ExecutionPlanSnapshot"
+        self,
```

Comment:
`_log_asset_planned_events` is a pretty long function already and it's getting significantly longer here. Is there a way to break it up a little?

Reply:
yup, split the materialization planned events logic out into a different function

Reply:
Consider putting the event generation on a staticmethod on `DagsterEvent`?

Reply:
yep, added